### ISSUE TITLE
css behavior - class binding not working and bootstrap classes cause issues

### ISSUE THREAD

**Issue by jpmyob** [ng-select/ng-select#254]
Please fill below information if issuing a bug report. 
### Expected behavior
The item should behave like any other form element for validations, of it is invalid and you class bind for display feedback I would expect the class to be applied. it is not.

Also - when applying classes that affect the ize of the box, an artifact (or hidden input) is visible.

Images included wit SO posting
https://stackoverflow.com/questions/48728271/ng-select-css-issues-bootstrap-input-sm

### Actual behaviour
So while using ng-select on a reactive form, I'm trying to style the field on error (invalid) like this.

the field:

<ng-select class="form-control input-sm" [class.store-error]="showError('store')"...

The show error method looks like this - and works for ALL OTHER FORM FIELDS

showError (fieldName: string) {
    const field = this.form.get(fieldName);
    return field.invalid ;
}

ng select does not change classes.

It appears as though ng-select constructs a div on top of a form field element, and changing the bootstrap class to input-sm exposes this (so a secondary question is how to get the input-sm applied to ng-select)

this CSS issue is secondary to the field NOT showing invalid using css-binding. Has anyone solved this?

### More Info
ng-select version: "@ng-select/ng-select": "^0.14.3",

browser:
FF 58.0.2 (64bit)
Chrome 64.0.3282.140 (Official Build) (64-bit)

reproducible in demo page:yes

**Comment by anjmao** [comment]
Hi @jpmyob, could you create simple repro step in issue plunker template https://plnkr.co/edit/hjZX6W?p=preview so I can look at it, I can't reproduce by myself. Thanks.

**Comment by jpmyob** [comment]
I'll do it tonight - thx for setting up a plunker

**Comment by jpmyob** [comment]
sorry for delay Anjmo - I'm working on recreating. , your example isn't using reactive forms, so Im getting issues setting up the environment like mine. if you add a button to chk form status to your example, and touch the select, withouth selecting one, the change focus, I bet you don't get a red 'error' border... but I'll get it working...to show you.

thanks for your patience

**Comment by jpmyob** [comment]
@anjmao - I was unable to replicate my form in your plunker, however, I found a solution, 

I needed to add a class to handle the reference.

This is the class used in site,

.state-error {
    background-color: #ffcccc;
    border-color: #ff0000;
}

This is the class I needed to add for the ng-select.

.ng-select.state-error .ng-control {
    background-color: #ffcccc;
    border-color: #ff0000;
}

All other things being equal on the validation front - this worked for me.
Thanks for responding - I hope this was in someway helpful
